### PR TITLE
Redesign `gr.Slider()`

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

Please include a concise summary, in clear English, of the changes in this pull request. If it closes an issue, please mention it here.

Closes: #8450
Closes: #937 

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by abidlabs** [comment]
Thanks @hannahblair. I agree that this slider feels a bit too bulky (the gray part that you slide the slider button across takes up too much vertical space). Also I'm not sure if its a good idea to put the number on the right slide as it reduces the horizontal space that the slider can slide across. I actually prefer the general layout of our current slider on those two points. 

My main UI/UX concerns with our current slider are:

* The slider color is blue instead of the "primary" color
* There's no way to reset to the original value of the design (https://github.com/gradio-app/gradio/issues/937)
* (Most annoyingly to me), the number input looks ugly and adds another border. Maybe the border should only appear if its being actively edited:

image

I think the `gr.Slider` doesn't need a significant design change if we are able to accommodate these issues, wdyt @hannahblair @dawoodkhan82 @pngwn?

**Review by abidlabs — APPROVED** [review]
Beautiful LGTM

**Comment by abidlabs** [comment]
Very small design nit: adding the minimum and maximum numbers is a nice touch, but it does end up eating into the real estate of the slider. Perhaps we reduce some of the horizontal padding between the sliders and numbers?

Screenshot 2024-08-29 at 1 00 01 PM

**Comment by abidlabs** [comment]
> @abidlabs do you reckon show_minmax would a useful param?

Yes, let's not add the parameter right now, but we could consider it if we get lots of user requests. 

>  alternatively they could do something like .min, .max { display: none } in some custom css

👍

**Comment by abidlabs** [comment]
Nice good call!

**Comment by rsxdalv** [comment]
Something got overlooked, I will see if I can find the solution to the visuals:

image

Maybe dark mode compatibility issue?

File: js/slider/Index.svelte [hunk 1]
```diff
@@ -128,106 +162,202 @@
...
+		background-color: var(--body-text-color-subdued);
 		cursor: not-allowed;
 	}
 
-	input[type="range"] {
-		-webkit-appearance: none;
-		appearance: none;
-		width: 100%;
-		accent-color: var(--slider-color);
-		height: 4px;
```
Comment (rsxdalv) [inline js/slider/Index.svelte:175]: height: 4px;
appears to fix it on Firefox
image

without:
image

So I am not getting a layout shift when it's not present. 

To be fair, a probably better option (perhaps alongside height) would be background: transparent or background that follows darkmode

Comment (rsxdalv) [reply js/slider/Index.svelte:175]: With the latest gradio 5.3.0 it seemed to be ok even on Firefox dark mode.

**Pull request merged**